operator/controllers/mlops/seldonruntime_controller.go

```diff
@@ -219,10 +225,10 @@ func (r *SeldonRuntimeReconciler) updateStatus(seldonRuntime *mlopsv1alpha1.Seld
 }
 
 // Find SeldonRuntimes that reference the changes SeldonConfig
-// TODO: pass an actual context from the caller to be used here
-func (r *SeldonRuntimeReconciler) mapSeldonRuntimesFromSeldonConfig(_ context.Context, obj client.Object) []reconcile.Request {
-	ctx, cancel := context.WithTimeout(context.Background(), constants.K8sAPICallsTxTimeout)
+func (r *SeldonRuntimeReconciler) mapSeldonRuntimesFromSeldonConfig(ctx context.Context, obj client.Object) []reconcile.Request {
+	ctx, cancel := context.WithTimeout(ctx, constants.K8sAPICallsTxTimeout)
```

Comment:
I notice that K8sAPICallsTxTimeout is quite large (2 minutes), with the main reasoning being that it is used for sequences of calls to the K8s API rather than individual calls. We should likely use a much smaller timeout (and a different constant) for single API calls (say, called `K8sAPICallTimeout`)